src/D2L.CodeStyle.Analyzers/Immutability/ImmutableTypeInfo.cs

```diff
@@ -1,34 +1,76 @@
 ﻿using System;
+using System.Collections.Immutable;
+using System.Linq;
 using Microsoft.CodeAnalysis;
 
 namespace D2L.CodeStyle.Analyzers.Immutability {
 	/// <summary>
 	/// A representation of the immutability of some type.
 	/// </summary>
 	internal readonly struct ImmutableTypeInfo : IEquatable<ImmutableTypeInfo> {
-		public ImmutableTypeInfo(
+
+		private ImmutableTypeInfo(
 			ImmutableTypeKind kind,
-			ITypeSymbol type
+			INamedTypeSymbol type,
+			ImmutableArray<bool> immutableTypeParameters
 		) {
 			Kind = kind;
 			Type = type;
+			ImmutableTypeParameters = immutableTypeParameters;
 		}
 
 		public ImmutableTypeKind Kind { get; }
 
-		public ITypeSymbol Type { get; }
+		public INamedTypeSymbol Type { get; }
+
+		public ImmutableArray<bool> ImmutableTypeParameters { get; }
+
+		public static ImmutableTypeInfo Create(
+			ImmutableTypeKind kind,
+			INamedTypeSymbol type
+		) {
+			ImmutableArray<bool> immutableTypeParameters = type
+				.TypeParameters
+				.Select( p => Attributes.Objects.Immutable.IsDefined( p ) )
+				.ToImmutableArray();
+
+			return new ImmutableTypeInfo(
+				kind: kind,
+				type: type,
+				immutableTypeParameters: immutableTypeParameters
+			);
+		}
+
+		public static ImmutableTypeInfo CreateWithAllImmutableTypeParameters(

```

Comment:
Imagine this would be used when loading in the "extra types". i.e we won't support adding an exception equivalent to `MyType<A, [Immutable] B>`

Reply:
I've successfully used this for value tuples.

Reply:
Oh nice, value tuple support would be good.